### ISSUE TITLE
Large force difference when set different symemtry_prec

### ISSUE THREAD

**Issue by pxlxingliang** [deepmodeling/abacus-develop#4739]
### Describe the bug

When I use default symmetry_prec (that is 1e-6) and symmetry_prec 1e-5, the forces are different:
symmetry_prec=1e-6:
```
------------------------------------------------------------------------------------------
 TOTAL-FORCE (eV/Angstrom)
------------------------------------------------------------------------------------------
                       Mo1         0.0000000000         0.0000000000         0.0000000000
                       Mo2         0.0000000000         2.1897877709         0.0000000000
                        S1         0.0000000000        -0.4863101120         0.0541911859
                        S2         0.0000000000        -0.4863101120        -0.0541911859
                        S3         0.0000000000         0.0000000000         0.0541911859
                        S4         0.0000000000         0.0000000000        -0.0541911859
------------------------------------------------------------------------------------------
```

symmetry_prec=1e-5
```
------------------------------------------------------------------------------------------
 TOTAL-FORCE (eV/Angstrom)
------------------------------------------------------------------------------------------
                       Mo1         0.0000000000         0.0000000000         0.0000000000
                       Mo2         0.0000000000         0.0000000000         0.0000000000
                        S1         0.0000000000         0.0000000000         0.0530687862
                        S2         0.0000000000         0.0000000000        -0.0530687862
                        S3         0.0000000000         0.0000000000         0.0530687862
                        S4         0.0000000000         0.0000000000        -0.0530687862
------------------------------------------------------------------------------------------
```

The force of QE:
```
     atom    1 type  1   force =     0.00000000   -0.00000000   -0.00000000
     atom    2 type  1   force =     0.00000000    0.00000000    0.00000000
     atom    3 type  2   force =     0.00000000    0.00000000    0.00195641
     atom    4 type  2   force =    -0.00000000    0.00000000   -0.00195641
     atom    5 type  2   force =     0.00000000   -0.00000000    0.00195641
     atom    6 type  2   force =    -0.00000000   -0.00000000   -0.00195641
```
[a (3).zip](https://github.com/user-attachments/files/16305268/a.3.zip)

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by pxlxingliang** [comment]
The example is from #4171

**Comment by WHUweiqingzhou** [comment]
Fixed by #5623
